replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/Worker.java

```diff
@@ -0,0 +1,354 @@
+      if (task != null) {
+        try {
+          task.unlock();
+        } catch (InterruptedException ignore) { // fallthrough
+        }
+      }
+      running = false;
+      processing = false;
+      throw e;
+    }
+  }
+
+  /**
+   * Set this workers running flag to false if it is not currently transferring an item
+   *
+   * @return {@code true} if the worker was cancelled, otherwise {@code false}
+   */
+  boolean cancelIfNotProcessing() {
+    if (!processing) {
+      this.running = false;
+      return true;
+    }
+    return false;
+  }
+
+  private boolean supported(Task task) {
+    if (!OperationType.HARVEST.equals(task.getOperation())) {
+      return false;
+    }
+
+    List<MetadataInfo> infos = task.metadatas().collect(Collectors.toList());
+    if (infos.size() > 1) {
+      LOGGER.warn("More than one metadata is currently not supported by this worker");
+      return false;
+    }
+
+    return (infos.stream().filter(DdfMetadataInfo.class::isInstance).count() == 1L);
+  }
+
+  @SuppressWarnings(
+      "squid:S3655" /* Optional.get() will return an info because we check it exists in the supported method */)
+  private void handle(Task task) throws InterruptedException {
+    NodeAdapter remote = getDestinationAdapter(task);
+    if (remote == null) {
+      task.unlock();
+      return;
+    }
+
+    try {
+      if (!adaptersAvailable(task, remote)) {
+        return;
+      }
+
+      DdfMetadataInfo info =
+          task.metadatas()
+              .filter(DdfMetadataInfo.class::isInstance)
+              .map(DdfMetadataInfo.class::cast)
+              .findFirst()
+              .get(); // should never end up being empty since it was filtered above
+
+      final Metadata metadata =
+          new MetadataImpl(
+              info.getData(), info.getDataClass(), task.getId(), Date.from(info.getLastModified()));
+
+      final OperationType type = task.getOperation();
+      if (type == OperationType.HARVEST) {
+        doCreate(task, metadata, remote, local);
+      } else {
+        // should never hit this since deserialization will return UNKNOWN
+        task.unlock();
+      }
+    } finally {
+      try {
+        remote.close();
+      } catch (IOException e) {
+        LOGGER.debug("Failed to close destination adapter", e);
+      }
+    }
+  }
+
+  @Nullable
+  private NodeAdapter getDestinationAdapter(Task task) {
+    Site site;
+    final String siteId = task.getQueue().getSite();
+    try {
+      site = siteManager.get(siteId);
+    } catch (NotFoundException e) {
+      LOGGER.debug("Unable to find site {} for queue", siteId, e);
+      return null;
+    } catch (ReplicationPersistenceException e) {
+      LOGGER.debug("Failed to fetch site {} for queue", siteId, e);
+      return null;
+    }
+
+    NodeAdapter nodeAdapter;
+    try {
+      nodeAdapter = nodeAdapters.factoryFor(site.getType()).create(site.getUrl());
+    } catch (IllegalArgumentException | AdapterException e) {
```

Comment:
❗️ Handle `AdapterInterruptedExcetion`

Reply:
The interface does not declare it, should it?

Reply:
probably better to do it since you are designing this to handle interruptions. On top of it, it should be the same way to handle it as the rest.